src/NVMLDevicePool.hpp

```diff
@@ -50,64 +50,79 @@ namespace geopm
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
-            virtual cpu_set_t *ideal_cpu_affinitization_mask(int accel_idx) const = 0;
+            /// @return CPU to Accelerator idea affinitization bitmask
+            virtual cpu_set_t *cpu_affinity_ideal_mask(int accel_idx) const = 0;
             /// @brief Get the NVML device streaming multiprocessor frequency
             //         in MHz
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator streaming multiproccesor frequency in MHz
             virtual uint64_t frequency_status_sm(int accel_idx) const = 0;
             /// @brief Get the NVML device utilization metric
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
-            virtual double utilization(int accel_idx) const = 0;
+            /// @return Accelerator streaming multiprocessor utilization
+            virtual uint64_t utilization(int accel_idx) const = 0;
             /// @brief Get the NVML device power in milliwatts
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator power consumption in milliwatts
             virtual uint64_t power (int accel_idx) const = 0;
             /// @brief Get the NVML device memory subsystem frequency in MHz
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator memory frequency in MHz
             virtual uint64_t frequency_status_mem(int accel_idx) const = 0;
             /// @brief Get the current NVML device clock throttle reasons
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator clock throttle reasons
             virtual uint64_t throttle_reasons(int accel_idx) const = 0;
             /// @brief Get the current NVML device temperature
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator temperature in Celsius
             virtual uint64_t temperature(int accel_idx) const = 0;
             /// @brief Get the total energy consumed counter value for
             //         an NVML device in millijoules
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator energy consumption in millijoules
             virtual uint64_t energy(int accel_idx) const = 0;
             /// @brief Get the current performance state of an NVML
             //         device
             /// @param [in] accel_idx The index indicating a particular
             ///        accelerator
+            /// @return Accelerator performance state
```

Comment:
Any more information about what a "performance state" means?

Reply:
The possible states are defined here: https://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceEnumvs.html#group__nvmlDeviceEnumvs_1g9de0cdfc67dfc2704b86344a7d5dc4fb

Non GPGPU description available here: 
https://docs.nvidia.com/gameworks/content/gameworkslibrary/coresdk/nvapi/group__gpupstate.html

The best guess from other documentation is that this ties to a preset min/max frequency range.